### PR TITLE
Accept Trace context to flow to out of proc languages

### PR DESCRIPTION
**Pull request by yojagad**
Part of fix for https://github.com/Azure/azure-functions-host/issues/3747

### CODE REVIEW

File: src/proto/FunctionRpc.proto [hunk 1]
```diff
@@ -255,6 +255,18 @@ message InvocationRequest {
 
   // binding metadata from trigger
   map<string, TypedData> trigger_metadata = 4;
+
+  // Populates activityId, tracestate and tags from host
+  TraceContext trace_context = 5;
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: rename to RpcTraceContext

File: src/proto/FunctionRpc.proto [hunk 2]
```diff
@@ -255,6 +255,18 @@ message InvocationRequest {
...
+}
+
+// Host sends ActivityId, traceStateString and Tags from host
+message TraceContext {
+	string Traceparent = 1;
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: add comments on each of the properties what is maps to on host. is Traceparent = ActivityId?

File: src/proto/FunctionRpc.proto [hunk 3]
```diff
@@ -255,6 +255,21 @@ message InvocationRequest {
...
   map<string, TypedData> trigger_metadata = 4;
+
+  // Populates activityId, tracestate and tags from host
+  RpcTraceContext trace_context = 5;
+}
+
+// Host sends ActivityId, traceStateString and Tags from host
+message RpcTraceContext {
+	// This corresponds to Activity.Current?.Id
+	string Traceparent = 1;
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: reaname to trace_parent, trace_state and attributes

File: src/proto/FunctionRpc.proto [hunk 4]
```diff
@@ -255,6 +255,21 @@ message InvocationRequest {
...
+// Host sends ActivityId, traceStateString and Tags from host
+message RpcTraceContext {
+	// This corresponds to Activity.Current?.Id
+	string trace_parent = 1;
+
+	// This corresponds to Activity.Current?.TraceStateString
+	string trace_state = 2;
+
+	// This corresponds to Activity.Current?.Tags
+	map<string, string> attributes = 3;
```
Comment (mhoeger) [inline src/proto/FunctionRpc.proto]: More context on the issues with map: https://github.com/Azure/azure-functions-language-worker-protobuf/issues/21

Comment (yojagad) [reply src/proto/FunctionRpc.proto]: Handled this on the host side

**Pull request merged**